pkg/btree/bptree_test.go

```diff
@@ -470,3 +471,98 @@ func TestBPTree_Iteration_SinglePage(t *testing.T) {
 		}
 	})
 }
+
+func TestBPTree_Iteration_FirstLast(t *testing.T) {
+	b := buftest.NewSeekableBuffer()
+	p, err := NewPageFile(b)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tree := NewBPTree(p, newTestMetaPage(t, p))
+	start := 10.0
+	increments := []float64{0.01, 0.05, 0.3}
+	currentIncrementIndex := 0
+
+	for i := start; i < 256; i += increments[currentIncrementIndex] {
+		buf := make([]byte, 8)
+		binary.LittleEndian.PutUint64(buf, math.Float64bits(i))
+
+		if err := tree.Insert(ReferencedValue{Value: buf}, MemoryPointer{Offset: uint64(i * 100), Length: uint32(len(buf))}); err != nil {
+			t.Fatal(err)
+		}
+
+		if int(i*100)%10 == 0 && currentIncrementIndex < len(increments)-1 {
+			currentIncrementIndex++
+		}
+	}
```

Comment:
to sort of mimic the dataset we're dealing with, I inserted a bunch of different floats, varying with decimal size. 

However, as we iterate, the value we insert should always be increasing